tests/test_backends.py

```diff
@@ -49,33 +52,99 @@ def test_get_backend_auto_detection_success_gguf(
     m_is_model_gguf.assert_called_once_with(tmp_gguf)
 
 
-# this test fails because the model_path is a directory but the platform is not linux
-@patch("sys.platform", "darwin")
-def test_get_backend_auto_detection_failure_vllm_dir_but_not_linux(
-    tmp_path: pathlib.Path,
+# this tests both cases where a valid and invalid safetensors model directory is supplied
+@pytest.mark.parametrize(
+    "safetensors_dir,expected",
+    [
+        (
+            "valid_safetensors_model_dir",
+            True,
+        ),
+        (
+            "invalid_safetensors_model_dir",
+            False,
+        ),
+    ],
+)
+def test_is_model_safetensors_valid(
+    safetensors_dir: str, expected: bool, testdata_path: pathlib.Path
 ):
-    with pytest.raises(ValueError) as exc_info:
-        backends.get(tmp_path, None)
-    assert "Cannot determine which backend to use" in str(exc_info.value)
+    test_json_dict = {"a": 1, "b": 2}
+    json_object = json.dumps(test_json_dict, indent=4)
+    safetensors_model_path = testdata_path / safetensors_dir
 
+    for file in ["config.json", "tokenizer.json", "tokenizer_config.json"]:
+        os.makedirs(os.path.dirname(safetensors_model_path / file), exist_ok=True)
+        with open(safetensors_model_path / file, "a+", encoding="UTF-8") as f:
+            f.write(json_object)
 
-# this test succeeds because the model_path is a directory and the platform is linux so vllm is
+    with open(
+        safetensors_model_path / "model.safetensors", "a+", encoding="UTF-8"
+    ) as f:
+        f.write("")
+    if expected:
+        with open(
+            safetensors_model_path / "tokenizer.model", "a+", encoding="UTF-8"
+        ) as f:
+            f.write("")
+
```

Comment:
The safetensors model path creation should get moved to a function. I see it being repeated in the first two new tests.

Reply:
done